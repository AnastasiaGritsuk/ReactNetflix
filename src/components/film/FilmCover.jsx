import * as React from "react";
import * as style from '../../styles/main.css';

export class Film extends React.Component {
    render() {
        return (
            <div className={style.filmWrapper}>
                <img src="" alt=""/>
                <div className={style.filmCover}>
                    <h4>Title</h4>
                    <p className={style.filmCoverSupHeader}></p>
                    <div className={style.filmCoverRatingIcon}></div>
                    <div>
                        <span>1994</span>
                        <span>154 min</span>
                    </div>
                    <p className={style.filmCoverDesc}>bla bla bla</p>
                    <p>director</p>
                    <p>cast list</p>
                </div>
            </div>
        );
    }
}