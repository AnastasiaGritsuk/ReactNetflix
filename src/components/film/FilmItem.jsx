import * as React from "react";
import * as style from '../../styles/main.css';
import { Link } from 'react-router-dom';

export class Item extends React.Component {
    render() {
        return (
            <div className={style.item}>
                <img src="" className={style.itemImg} alt=""/>
                <div className={style.itemCaptionWrap}>
                    <div className={style.itemCaption}>
                        <a className={style.itemTitle}>
                            <Link to="/film">{this.props.film.title}}</Link>
                        </a>
                        <span className={style.itemTitleYear}>{this.props.film.releaseYear}</span>
                    </div>
                    <p className={style.textNote}>{this.props.film.genre}</p>
                </div>
            </div>
        );
    }
}