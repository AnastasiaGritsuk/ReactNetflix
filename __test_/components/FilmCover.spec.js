import React from 'react';
import { FilmCover } from '../../src/components/film/FilmCover';
import renderer from 'react-test-renderer';

it('Should show movie info', () => {
    let props = {
        film: {
            id: 1, title: 'Film test',
            releaseYear: 2017,
            genre: 'name test',
            director: 'director test',
            rating: '5',
            duration: 120
        }
    };
    const component = renderer.create(
        <FilmCover {...props}></FilmCover>,
    );
    let tree = component.toJSON();
    expect(tree).toMatchSnapshot();
});