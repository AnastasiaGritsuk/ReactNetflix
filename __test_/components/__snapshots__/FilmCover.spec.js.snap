// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Should show movie info 1`] = `
<div>
  <img
    alt=""
    src=""
  />
  <div>
    <h4>
      Title 
      Film test
    </h4>
    <p />
    <div>
      Rating 
      5
    </div>
    <div>
      <span>
        1994 
        2017
      </span>
      <span>
        154 min 
        120
      </span>
    </div>
    <p>
      bla bla bla 
    </p>
    <p>
      director 
      director test
    </p>
    <p>
      cast list
    </p>
  </div>
</div>
`;
